tests/units/classes/Project/Qgis/VectorLayerEditWidgetTest.php

```diff
@@ -300,4 +301,325 @@ public function testRangeFromXmlReader(): void
             $this->assertFalse(isset($editWidget->config->{$not}), $not);
         }
     }
+
+    public function testValueRelationFromXmlReader(): void
+    {
+        // No filter Expression
+        $xmlStr = '
+        <editWidget type="ValueRelation">
+          <config>
+            <Option type="Map">
+              <Option type="bool" value="false" name="AllowMulti"/>
+              <Option value="true" name="AllowNull" type="bool"/>
+              <Option name="FilterExpression" type="invalid"/>
+              <Option value="es_id" name="Key" type="QString"/>
+              <Option value="espece20160621120237434" name="Layer" type="QString"/>
+              <Option value="espece" name="LayerName" type="QString"/>
+              <Option value="postgres" name="LayerProviderName" type="QString"/>
+              <Option value="service=\'PG_SERVICE\' sslmode=disable key=\'es_id\' estimatedmetadata=true checkPrimaryKeyUnicity=\'1\' table=&quot;observatoire&quot;.&quot;espece&quot; sql=" name="LayerSource" type="QString"/>
+              <Option value="1" name="NofColumns" type="int"/>
+              <Option value="true" name="OrderByValue" type="bool"/>
+              <Option value="false" name="UseCompleter" type="bool"/>
+              <Option value="es_nom_commun" name="Value" type="QString"/>
+            </Option>
+          </config>
+        </editWidget>
+        ';
+        $oXml = App\XmlTools::xmlReaderFromString($xmlStr);
+        $editWidget = VectorLayerEditWidget::fromXmlReader($oXml);
+
+        $this->assertEquals('ValueRelation', $editWidget->type);
+        $this->assertNotNull($editWidget->config);
+        $this->assertInstanceOf(EditWidget\ValueRelationConfig::class, $editWidget->config);
+
+        $config = array(
+            'AllowMulti' => false,
+            'AllowNull' => true,
+            'FilterExpression' => '',
+            'Key' => 'es_id',
+            'Layer' => 'espece20160621120237434',
+            'LayerName' => 'espece',
+            'LayerProviderName' => 'postgres',
+            'NofColumns' => 1,
+            'OrderByValue' => true,
+            'UseCompleter' => false,
+            'Value' => 'es_nom_commun',
+        );
+        foreach ($config as $prop => $value) {
+            $this->assertSame($value, $editWidget->config->{$prop}, $prop);
+        }
+
+        // Old Value relation
```

Comment:
What is "Old" ?
Can you add a string like "before QGIS 3.40".

I would be nice to have it in other places.

This is for, in 2 or 3 years, to know if this code can be safely removed because QGIS 3.34 will not be supported and we can do some cleaning into the code about "OLD" tests.

Thanks